src/olympia/addons/tests/test_views.py

```diff
@@ -1364,6 +1365,107 @@ def test_set_tags(self):
 
+    def test_default_locale_with_invalid_locale(self):
+        response = self.client.post(
+            self.url,
+            data={**self.minimal_data, 'default_locale': 'zz'},
+        )
+        assert response.status_code == 400
+        assert response.data == {'default_locale': ['"zz" is not a valid choice.']}
+
+    def test_default_locale(self):
+        # An xpi without localization:
+        # failure cases:
+        # field has other translations provided
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'default_locale': 'fr',
+                'description': {'de': 'Das description'},
+            },
+        )
+        assert response.status_code == 400, response.data
+        error_string = 'A value in the default locale of "fr" is required.'
+        assert response.data == {
+            'description': [error_string],
+        }
+
+        # success cases:
+        # field is provided with a value in new default
+        # field already has a value in new default
+        # field has no other translations
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'default_locale': 'fr',
+                'name': {'fr': 'nom française'},
+                # no summary, but does have a value in the manifest already
+                # no description and doesn't have other translations
+            },
+        )
+        assert response.status_code == 201, response.data
+        addon = Addon.objects.get()
+        assert addon.default_locale == 'fr'
+        assert addon.name == 'nom française'
+        assert addon.name.locale == 'fr'
+        assert addon.summary == 'just a test addon with the manifest.json format'
+        assert addon.summary.locale == 'fr'
+        assert addon.description is None
+
+    def test_default_locale_localized_xpi(self):
+        upload = self.get_upload(
+            'notify-link-clicks-i18n-missing.xpi',
+            user=self.user,
+            source=amo.UPLOAD_SOURCE_ADDON_API,
+            channel=amo.RELEASE_CHANNEL_UNLISTED,
+        )
+
+        error_string = 'A value in the default locale of "de" is required.'
+        # field doesn't have a value in the xpi in new default, or
+        # field has other translations provided
+        response = self.client.post(
+            self.url,
+            data={
+                'version': {'upload': upload.uuid},
+                'default_locale': 'de',
+                # 'name' - not provided in de in the xpi
+                'support_url': {'it': 'https://it.support.test/'},
+            },
+        )
+        assert response.status_code == 400, response.data
+        assert response.data == {
+            'name': [error_string],
+            'support_url': [error_string],
+        }
+
+        # success cases:
+        # field is provided with a value in new default
```

Comment:
Same question here.

Reply:
same here, the only difference is `summary` has a localized value in the _locales files in the xpi in `de` rather than just a string directly in the manifest.